### PR TITLE
Improve Swagger's DocumentationNormalizer

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Swagger/Serializer/DocumentationNormalizer.php [hunk 1]
```diff
@@ -132,195 +132,109 @@ private function getPathOperation(string $operationName, array $operation, strin
...
-                            'schema' => [
-                                '$ref' => sprintf('#/definitions/%s', $resourceShortName),
-                            ],
-                        ],
-                        '400' => [
-                            'description' => 'Invalid input',
-                        ],
-                        '404' => [
-                            'description' => 'Resource not found',
+        $pathOperation['tags'] ?? $pathOperation['tags'] = [$resourceShortName];
```
Comment (teohhanhui) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: What's going on here? :laughing:

Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: Git/GitHub fails to create a good dist, but the underlying code looks ok.

Comment (teohhanhui) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: I mean, shouldn't it be:

```php
$pathOperation['tags'] = $pathOperation['tags'] ?? [$resourceShortName];
```

Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: Why? It makes an useless assignation.

Comment (teohhanhui) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: IMO that's a bad coding style. Too cryptic and most likely doesn't even have any practical benefit.

It's short-circuit on steroids... lol

Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: I find it clearer but it's probably a personal taste :)

Comment (soyuka) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: It's called job-safety not short-circuit on steroids :smile_cat:

File: src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php [hunk 2]
```diff
@@ -129,8 +133,8 @@ private function parseResource(ResourceMetadata $resourceMetadata, string $resou
     private function getGroupsForItemAndCollectionOperation(ResourceMetadata $resourceMetadata, string $operationName) : array
     {
         $operation = [
-            'denormalization_context' => array_merge($resourceMetadata->getItemOperationAttribute($operationName, 'denormalization_context', []), $resourceMetadata->getCollectionOperationAttribute($operationName, 'denormalization_context', [])),
-            'normalization_context' => array_merge($resourceMetadata->getItemOperationAttribute($operationName, 'normalization_context', []), $resourceMetadata->getCollectionOperationAttribute($operationName, 'normalization_context', [])),
+            'denormalization_context' => $resourceMetadata->getItemOperationAttribute($operationName, 'denormalization_context', []) + $resourceMetadata->getCollectionOperationAttribute($operationName, 'denormalization_context', []),
```
Comment (teohhanhui) [inline src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php]: Why is there union of item + collection operation attributes going on? Only one of them can be applicable at a time.

Even if this makes any sense and is the correct thing to do, what about collection and item operations with the same name? Too risky...

File: src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php [hunk 3]
```diff
@@ -129,8 +133,8 @@ private function parseResource(ResourceMetadata $resourceMetadata, string $resou
     private function getGroupsForItemAndCollectionOperation(ResourceMetadata $resourceMetadata, string $operationName) : array
```
Comment (teohhanhui) [inline src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php]: Actually, the logic related to this whole function needs to be reviewed. I gave it a quick look and things look very fishy here.

File: src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php (hunk 2 above)
Comment (dunglas) [inline src/Bridge/NelmioApiDoc/Parser/ApiPlatformParser.php]: I 100% agree, this must be investigated. But this PR or doesn't change the behavior, it's just a style fix. I propose to refactor this method in another PR. WDYT?

**Pull request merged**